create.go

```diff
@@ -142,17 +142,39 @@ func (r *Resource) computePatches(ctx context.Context, accessor metav1.Object, o
 	// might be other properties managed by external authorities who have to
 	// manage their own initialization.
 	{
-		if clusterStatus.Conditions == nil && clusterStatus.Versions == nil {
+		conditionsEmpty := clusterStatus.Conditions == nil
+		nodesEmpty := clusterStatus.Nodes == nil
+		versionsEmpty := clusterStatus.Versions == nil
+
+		if conditionsEmpty && nodesEmpty && versionsEmpty {
+			patches = append(patches, Patch{
+				Op:    "add",
+				Path:  "/status",
+				Value: Status{},
+			})
+		}
+
+		if conditionsEmpty {
+			patches = append(patches, Patch{
+				Op:    "replace",
+				Path:  "/status/cluster/conditions",
+				Value: []providerv1alpha1.StatusClusterCondition{},
+			})
+		}
```

Comment:
I want the defaulting to be more reliable and dynamic so this is a bit separated now and every status property managed by the resource is managed in its own dedicated way based on its own assumptions.